### ISSUE TITLE
Stuck sending first query in Linux subsystem (Windows 10)

### ISSUE THREAD

**Issue by nathan130200** [Levak/warfacebot#105]
Just it appears in console window:

```bash
localhost@DESKTOP-B9F5EHE:/mnt/c/Users/natha/Documents/CLion/Projects/warfacebot$ ./wb.sh br
Login: <login>
Password: <password>
Connecting...done

Token Generated is <token>
g_language = english
game_hwid = 0
wb_accept_friend_requests = 1
wb_postpone_friend_requests = 0
wb_accept_clan_invites = 1
wb_postpone_clan_invites = 0
wb_enable_invite = 1
online_server = game.warface.levelupgames.com.br
game_version = 1.15000.789.36300
game_server_name = br

Warfacebot Copyright (C) 2015-2017 Levak Borok
This program comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions; see AGPLv3 Terms for details.


Loading cached `items'
Cannot open file for reading: ./QueryCache/items.xml
Loading cached `shop_get_offers'
Cannot open file for reading: ./QueryCache/shop_get_offers.xml
Loading cached `quickplay_maplist'
Cannot open file for reading: ./QueryCache/quickplay_maplist.xml
Loading cached `get_configs'
Cannot open file for reading: ./QueryCache/get_configs.xml

CMD# ^C
CMD#
Closed idle
##--(131)-> <iq to='k01.warface' type='get'><query xmlns='urn:cryonline:k01'><player_status prev_status='0' new_status='2' to=''/></query></iq>
##--( 16)-> </stream:stream>
Warface Bot closed!
```

I compiled using Ubuntu Subsystem for Windows 10. I Followed all steps like .travis.yml all packages are installed correctly.

Don't make any requests. :(

Tried: 
`$MAKEFILE_RULE=debug-wb`
`make -j 4`

And:
`make debug-wb -j 4`

In (131) i pressed CTRL+D to terminate

**Comment by Levak** [comment]
Try `make clean debug-wb -j 4`.
When switching between debug/non-debug rules, there is no recompilation imposed by build system, even thought it needs to recompile the object files.

**Comment by FRNathan13** [comment]
After execute i receive this message `./wb.sh: linha 231: ./wb: File or directory not found`

**Comment by FRNathan13** [comment]
```bash
localhost@DESKTOP-B9F5EHE:/mnt/c/Users/natha/Documents/CLion/Projects/warfacebot$ make clean debug-wb -j 4
rm -f wb wbs ./src/clanmate.o ./src/cmd/complete.o ./src/cmd/cmd_add.o ./src/cmd/cmd_change.o ./src/cmd/cmd_channel.o ./src/cmd/cmd_exec.o ./src/cmd/cmd_follow.o ./src/cmd/cmd_friends.o ./src/cmd/cmd_invite.o ./src/cmd/cmd_last.o ./src/cmd/cmd_leave.o ./src/cmd/cmd_master.o ./src/cmd/cmd_missions.o ./src/cmd/cmd_name.o ./src/cmd/cmd_open.o ./src/cmd/cmd_quit.o ./src/cmd/cmd_randombox.o ./src/cmd/cmd_ready.o ./src/cmd/cmd_remove.o ./src/cmd/cmd_safe.o ./src/cmd/cmd_say.o ./src/cmd/cmd_sleep.o ./src/cmd/cmd_sponsor.o ./src/cmd/cmd_start.o ./src/cmd/cmd_stats.o ./src/cmd/cmd_stay.o ./src/cmd/cmd_switch.o ./src/cmd/cmd_unready.o ./src/cmd/cmd_whisper.o ./src/cmd/cmd_whois.o ./src/cmd/cmd_quickplay.o ./src/cvar.o ./src/friend.o ./src/gameroom.o ./src/item.o ./src/lang.o ./src/offer.o ./src/option.o ./src/masterserver.o ./src/mission.o ./src/querycache.o ./src/querydump.o ./src/quickplay.o ./src/pvp_maps.o ./src/session.o ./src/status.o ./src/stream/connect.o ./src/stream/crypt.o ./src/stream/recv.o ./src/stream/send.o ./src/stream/tls.o ./src/threads/threads.o ./src/threads/thread_dispatch.o ./src/threads/thread_ping.o ./src/threads/thread_sendstream.o ./src/threads/thread_readstream.o ./src/threads/thread_readline.o ./third_party/rl_fprintf.o ./third_party/stub-getline.o ./third_party/stub-strcasestr.o ./third_party/stub-strndup.o ./third_party/stub-strtok_r.o ./third_party/wake.o ./src/tools/base64.o ./src/tools/geoip.o ./src/tools/get_info.o ./src/tools/list.o ./src/tools/uuid.o ./src/tools/zlibb64.o ./src/wb.o ./src/xml/serializer.o ./src/xmpp/get/xmpp_bind.o ./src/xmpp/get/xmpp_iq_ping.o ./src/xmpp/get/xmpp_iq_session.o ./src/xmpp/get/xmpp_presence.o ./src/xmpp/get/xmpp_room_message.o ./src/xmpp/get/xmpp_sasl.o ./src/xmpp/get/xmpp_starttls.o ./src/xmpp/get/xmpp_stream.o ./src/xmpp/id_handler.o ./src/xmpp/query_handler.o ./src/xmpp/result/xmpp_iq_ping.o ./src/xmpp/result/xmpp_presence.o ./src/xmpp/room.o ./src/xmpp/sasl.o ./src/xmpp/send.o ./src/xmpp/tools.o ./src/xmpp_wf/close.o ./src/xmpp_wf/connect.o ./src/xmpp_wf/get/xmpp_iq_account.o ./src/xmpp_wf/get/xmpp_iq_confirm_notification.o ./src/xmpp_wf/get/xmpp_iq_create_profile.o ./src/xmpp_wf/get/xmpp_iq_follow_send.o ./src/xmpp_wf/get/xmpp_iq_gameroom_askserver.o ./src/xmpp_wf/get/xmpp_iq_gameroom_get.o ./src/xmpp_wf/get/xmpp_iq_gameroom_join.o ./src/xmpp_wf/get/xmpp_iq_gameroom_kick.o ./src/xmpp_wf/get/xmpp_iq_gameroom_leave.o ./src/xmpp_wf/get/xmpp_iq_gameroom_open.o ./src/xmpp_wf/get/xmpp_iq_gameroom_quickplay.o ./src/xmpp_wf/get/xmpp_iq_gameroom_quickplay_cancel.o ./src/xmpp_wf/get/xmpp_iq_preinvite_cancel.o ./src/xmpp_wf/get/xmpp_iq_preinvite_invite.o ./src/xmpp_wf/get/xmpp_iq_gameroom_promote_to_host.o ./src/xmpp_wf/get/xmpp_iq_gameroom_setinfo.o ./src/xmpp_wf/get/xmpp_iq_gameroom_setname.o ./src/xmpp_wf/get/xmpp_iq_gameroom_setplayer.o ./src/xmpp_wf/get/xmpp_iq_gameroom_update_pvp.o ./src/xmpp_wf/get/xmpp_iq_get_account_profiles.o ./src/xmpp_wf/get/xmpp_iq_get_achievements.o ./src/xmpp_wf/get/xmpp_iq_get_configs.o ./src/xmpp_wf/get/xmpp_iq_get_last_seen_date.o ./src/xmpp_wf/get/xmpp_iq_get_master_servers.o ./src/xmpp_wf/get/xmpp_iq_get_master_server.o ./src/xmpp_wf/get/xmpp_iq_get_player_stats.o ./src/xmpp_wf/get/xmpp_iq_invitation_send.o ./src/xmpp_wf/get/xmpp_iq_items.o ./src/xmpp_wf/get/xmpp_iq_join_channel.o ./src/xmpp_wf/get/xmpp_iq_missions_get_list.o ./src/xmpp_wf/get/xmpp_iq_peer_clan_member_update.o ./src/xmpp_wf/get/xmpp_iq_peer_player_info.o ./src/xmpp_wf/get/xmpp_iq_peer_status_update.o ./src/xmpp_wf/get/xmpp_iq_player_status.o ./src/xmpp_wf/get/xmpp_iq_profile_info_get_status.o ./src/xmpp_wf/get/xmpp_iq_querycache.o ./src/xmpp_wf/get/xmpp_iq_quickplay_maplist.o ./src/xmpp_wf/get/xmpp_iq_remove_friend.o ./src/xmpp_wf/get/xmpp_iq_resync_profile.o ./src/xmpp_wf/get/xmpp_iq_shop_get_offers.o ./src/xmpp_wf/get/xmpp_iq_send_invitation.o ./src/xmpp_wf/get/xmpp_print_number_of_occupants.o ./src/xmpp_wf/get/xmpp_send_message.o ./src/xmpp_wf/result/xmpp_iq_autorotate.o ./src/xmpp_wf/result/xmpp_iq_broadcast_session_result.o ./src/xmpp_wf/result/xmpp_iq_clan_info.o ./src/xmpp_wf/result/xmpp_iq_clan_masterbanner_update.o ./src/xmpp_wf/result/xmpp_iq_clan_members_updated.o ./src/xmpp_wf/result/xmpp_iq_follow_send.o ./src/xmpp_wf/result/xmpp_iq_friend_list.o ./src/xmpp_wf/result/xmpp_iq_gameroom_loosemaster.o ./src/xmpp_wf/result/xmpp_iq_gameroom_offer.o ./src/xmpp_wf/result/xmpp_iq_gameroom_on_expired.o ./src/xmpp_wf/result/xmpp_iq_gameroom_on_kicked.o ./src/xmpp_wf/result/xmpp_iq_gameroom_quickplay_started.o ./src/xmpp_wf/result/xmpp_iq_gameroom_quickplay_succeeded.o ./src/xmpp_wf/result/xmpp_iq_gameroom_quickplay_canceled.o ./src/xmpp_wf/result/xmpp_iq_gameroom_sync.o ./src/xmpp_wf/result/xmpp_iq_invitation_request.o ./src/xmpp_wf/result/xmpp_iq_invitation_result.o ./src/xmpp_wf/result/xmpp_iq_notification_broadcast.o ./src/xmpp_wf/result/xmpp_iq_p2p_ping.o ./src/xmpp_wf/result/xmpp_iq_peer_clan_member_update.o ./src/xmpp_wf/result/xmpp_iq_peer_player_info.o ./src/xmpp_wf/result/xmpp_iq_peer_status_update.o ./src/xmpp_wf/result/xmpp_iq_preinvite_cancel.o ./src/xmpp_wf/result/xmpp_iq_preinvite_invite.o ./src/xmpp_wf/result/xmpp_iq_preinvite_response.o ./src/xmpp_wf/result/xmpp_iq_shop_sync_consummables.o ./src/xmpp_wf/result/xmpp_iq_sponsor_info_updated.o ./src/xmpp_wf/result/xmpp_iq_sync_notifications.o ./src/xmpp_wf/result/xmpp_iq_update_cry_money.o ./src/xmpp_wf/result/xmpp_message.o ./src/xmpp_wf/tools.o
make: *** No rule to make target 'debug-wb'.  Pare.
make: ** Esperando que outros processos terminem.
rm -f wbm wbd generated-wb-api.[cho] ./src/dbus/wbm.o ./src/dbus/wbd.o ./src/dbus/methods/buddies.o ./src/dbus/methods/buddy_add.o ./src/dbus/methods/buddy_follow.o ./src/dbus/methods/buddy_invite.o ./src/dbus/methods/buddy_last_seen.o ./src/dbus/methods/buddy_remove.o ./src/dbus/methods/buddy_whisper.o ./src/dbus/methods/buddy_whois.o ./src/dbus/methods/channel_stats.o ./src/dbus/methods/channel_switch.o ./src/dbus/methods/chat_room_join.o ./src/dbus/methods/chat_room_leave.o ./src/dbus/methods/crown_challenge.o ./src/dbus/methods/quit.o ./src/dbus/methods/room_change_map.o ./src/dbus/methods/room_change_team.o ./src/dbus/methods/room_give_master.o ./src/dbus/methods/room_leave.o ./src/dbus/methods/room_open.o ./src/dbus/methods/room_participants.o ./src/dbus/methods/room_ready.o ./src/dbus/methods/room_rename.o ./src/dbus/methods/room_say.o ./src/dbus/methods/room_start.o ./src/dbus/methods/room_take_class.o generated-wb-api.o
rm -f .opt ./src/clanmate.d ./src/cmd/complete.d ./src/cmd/cmd_add.d ./src/cmd/cmd_change.d ./src/cmd/cmd_channel.d ./src/cmd/cmd_exec.d ./src/cmd/cmd_follow.d ./src/cmd/cmd_friends.d ./src/cmd/cmd_invite.d ./src/cmd/cmd_last.d ./src/cmd/cmd_leave.d ./src/cmd/cmd_master.d ./src/cmd/cmd_missions.d ./src/cmd/cmd_name.d ./src/cmd/cmd_open.d ./src/cmd/cmd_quit.d ./src/cmd/cmd_randombox.d ./src/cmd/cmd_ready.d ./src/cmd/cmd_remove.d ./src/cmd/cmd_safe.d ./src/cmd/cmd_say.d ./src/cmd/cmd_sleep.d ./src/cmd/cmd_sponsor.d ./src/cmd/cmd_start.d ./src/cmd/cmd_stats.d ./src/cmd/cmd_stay.d ./src/cmd/cmd_switch.d ./src/cmd/cmd_unready.d ./src/cmd/cmd_whisper.d ./src/cmd/cmd_whois.d ./src/cmd/cmd_quickplay.d ./src/cvar.d ./src/friend.d ./src/gameroom.d ./src/item.d ./src/lang.d ./src/offer.d ./src/option.d ./src/masterserver.d ./src/mission.d ./src/querycache.d ./src/querydump.d ./src/quickplay.d ./src/pvp_maps.d ./src/session.d ./src/status.d ./src/stream/connect.d ./src/stream/crypt.d ./src/stream/recv.d ./src/stream/send.d ./src/stream/tls.d ./src/threads/threads.d ./src/threads/thread_dispatch.d ./src/threads/thread_ping.d ./src/threads/thread_sendstream.d ./src/threads/thread_readstream.d ./src/threads/thread_readline.d ./third_party/rl_fprintf.d ./third_party/stub-getline.d ./third_party/stub-strcasestr.d ./third_party/stub-strndup.d ./third_party/stub-strtok_r.d ./third_party/wake.d ./src/tools/base64.d ./src/tools/geoip.d ./src/tools/get_info.d ./src/tools/list.d ./src/tools/uuid.d ./src/tools/zlibb64.d ./src/wb.d ./src/xml/serializer.d ./src/xmpp/get/xmpp_bind.d ./src/xmpp/get/xmpp_iq_ping.d ./src/xmpp/get/xmpp_iq_session.d ./src/xmpp/get/xmpp_presence.d ./src/xmpp/get/xmpp_room_message.d ./src/xmpp/get/xmpp_sasl.d ./src/xmpp/get/xmpp_starttls.d ./src/xmpp/get/xmpp_stream.d ./src/xmpp/id_handler.d ./src/xmpp/query_handler.d ./src/xmpp/result/xmpp_iq_ping.d ./src/xmpp/result/xmpp_presence.d ./src/xmpp/room.d ./src/xmpp/sasl.d ./src/xmpp/send.d ./src/xmpp/tools.d ./src/xmpp_wf/close.d ./src/xmpp_wf/connect.d ./src/xmpp_wf/get/xmpp_iq_account.d ./src/xmpp_wf/get/xmpp_iq_confirm_notification.d ./src/xmpp_wf/get/xmpp_iq_create_profile.d ./src/xmpp_wf/get/xmpp_iq_follow_send.d ./src/xmpp_wf/get/xmpp_iq_gameroom_askserver.d ./src/xmpp_wf/get/xmpp_iq_gameroom_get.d ./src/xmpp_wf/get/xmpp_iq_gameroom_join.d ./src/xmpp_wf/get/xmpp_iq_gameroom_kick.d ./src/xmpp_wf/get/xmpp_iq_gameroom_leave.d ./src/xmpp_wf/get/xmpp_iq_gameroom_open.d ./src/xmpp_wf/get/xmpp_iq_gameroom_quickplay.d ./src/xmpp_wf/get/xmpp_iq_gameroom_quickplay_cancel.d ./src/xmpp_wf/get/xmpp_iq_preinvite_cancel.d ./src/xmpp_wf/get/xmpp_iq_preinvite_invite.d ./src/xmpp_wf/get/xmpp_iq_gameroom_promote_to_host.d ./src/xmpp_wf/get/xmpp_iq_gameroom_setinfo.d ./src/xmpp_wf/get/xmpp_iq_gameroom_setname.d ./src/xmpp_wf/get/xmpp_iq_gameroom_setplayer.d ./src/xmpp_wf/get/xmpp_iq_gameroom_update_pvp.d ./src/xmpp_wf/get/xmpp_iq_get_account_profiles.d ./src/xmpp_wf/get/xmpp_iq_get_achievements.d ./src/xmpp_wf/get/xmpp_iq_get_configs.d ./src/xmpp_wf/get/xmpp_iq_get_last_seen_date.d ./src/xmpp_wf/get/xmpp_iq_get_master_servers.d ./src/xmpp_wf/get/xmpp_iq_get_master_server.d ./src/xmpp_wf/get/xmpp_iq_get_player_stats.d ./src/xmpp_wf/get/xmpp_iq_invitation_send.d ./src/xmpp_wf/get/xmpp_iq_items.d ./src/xmpp_wf/get/xmpp_iq_join_channel.d ./src/xmpp_wf/get/xmpp_iq_missions_get_list.d ./src/xmpp_wf/get/xmpp_iq_peer_clan_member_update.d ./src/xmpp_wf/get/xmpp_iq_peer_player_info.d ./src/xmpp_wf/get/xmpp_iq_peer_status_update.d ./src/xmpp_wf/get/xmpp_iq_player_status.d ./src/xmpp_wf/get/xmpp_iq_profile_info_get_status.d ./src/xmpp_wf/get/xmpp_iq_querycache.d ./src/xmpp_wf/get/xmpp_iq_quickplay_maplist.d ./src/xmpp_wf/get/xmpp_iq_remove_friend.d ./src/xmpp_wf/get/xmpp_iq_resync_profile.d ./src/xmpp_wf/get/xmpp_iq_shop_get_offers.d ./src/xmpp_wf/get/xmpp_iq_send_invitation.d ./src/xmpp_wf/get/xmpp_print_number_of_occupants.d ./src/xmpp_wf/get/xmpp_send_message.d ./src/xmpp_wf/result/xmpp_iq_autorotate.d ./src/xmpp_wf/result/xmpp_iq_broadcast_session_result.d ./src/xmpp_wf/result/xmpp_iq_clan_info.d ./src/xmpp_wf/result/xmpp_iq_clan_masterbanner_update.d ./src/xmpp_wf/result/xmpp_iq_clan_members_updated.d ./src/xmpp_wf/result/xmpp_iq_follow_send.d ./src/xmpp_wf/result/xmpp_iq_friend_list.d ./src/xmpp_wf/result/xmpp_iq_gameroom_loosemaster.d ./src/xmpp_wf/result/xmpp_iq_gameroom_offer.d ./src/xmpp_wf/result/xmpp_iq_gameroom_on_expired.d ./src/xmpp_wf/result/xmpp_iq_gameroom_on_kicked.d ./src/xmpp_wf/result/xmpp_iq_gameroom_quickplay_started.d ./src/xmpp_wf/result/xmpp_iq_gameroom_quickplay_succeeded.d ./src/xmpp_wf/result/xmpp_iq_gameroom_quickplay_canceled.d ./src/xmpp_wf/result/xmpp_iq_gameroom_sync.d ./src/xmpp_wf/result/xmpp_iq_invitation_request.d ./src/xmpp_wf/result/xmpp_iq_invitation_result.d ./src/xmpp_wf/result/xmpp_iq_notification_broadcast.d ./src/xmpp_wf/result/xmpp_iq_p2p_ping.d ./src/xmpp_wf/result/xmpp_iq_peer_clan_member_update.d ./src/xmpp_wf/result/xmpp_iq_peer_player_info.d ./src/xmpp_wf/result/xmpp_iq_peer_status_update.d ./src/xmpp_wf/result/xmpp_iq_preinvite_cancel.d ./src/xmpp_wf/result/xmpp_iq_preinvite_invite.d ./src/xmpp_wf/result/xmpp_iq_preinvite_response.d ./src/xmpp_wf/result/xmpp_iq_shop_sync_consummables.d ./src/xmpp_wf/result/xmpp_iq_sponsor_info_updated.d ./src/xmpp_wf/result/xmpp_iq_sync_notifications.d ./src/xmpp_wf/result/xmpp_iq_update_cry_money.d ./src/xmpp_wf/result/xmpp_message.d ./src/xmpp_wf/tools.d ./src/dbus/wbd.d ./src/dbus/methods/buddies.d ./src/dbus/methods/buddy_add.d ./src/dbus/methods/buddy_follow.d ./src/dbus/methods/buddy_invite.d ./src/dbus/methods/buddy_last_seen.d ./src/dbus/methods/buddy_remove.d ./src/dbus/methods/buddy_whisper.d ./src/dbus/methods/buddy_whois.d ./src/dbus/methods/channel_stats.d ./src/dbus/methods/channel_switch.d ./src/dbus/methods/chat_room_join.d ./src/dbus/methods/chat_room_leave.d ./src/dbus/methods/crown_challenge.d ./src/dbus/methods/quit.d ./src/dbus/methods/room_change_map.d ./src/dbus/methods/room_change_team.d ./src/dbus/methods/room_give_master.d ./src/dbus/methods/room_leave.d ./src/dbus/methods/room_open.d ./src/dbus/methods/room_participants.d ./src/dbus/methods/room_ready.d ./src/dbus/methods/room_rename.d ./src/dbus/methods/room_say.d ./src/dbus/methods/room_start.d ./src/dbus/methods/room_take_class.d generated-wb-api.d
localhost@DESKTOP-B9F5EHE:/mnt/c/Users/natha/Documents/CLion/Projects/warfacebot$ sudo ./wb.sh br
Login: <login>
Password: <password>
Connecting...done

Token Generated is <token>
./wb.sh: linha 231: ./wb: Arquivo ou diretório não encontrado
localhost@DESKTOP-B9F5EHE:/mnt/c/Users/natha/Documents/CLion/Projects/warfacebot$```

**Comment by FRNathan13** [comment]
I will try clone repo again.

**Comment by FRNathan13** [comment]
I executed `make clean debug-wb -j 4` same. Stop at 
```xml
##--(148)-> <?xml version='1.0' ?><stream:stream to='warface' xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' xml:lang='en' version='1.0'>```

**Comment by Levak** [comment]
Then they changed the DNS for the lobby server.
Or you are IP banned.

Try `ping game.warface.levelupgames.com.br`

**Comment by FRNathan13** [comment]
Timeout

**Comment by FRNathan13** [comment]
Send 4, Receive 0, Lost 4 (100% Loss)

**Comment by FRNathan13** [comment]
I'm playing normally, in this moment warface is open on my seconds account.

**Comment by Levak** [comment]
Please, stop posting 1 message a second. You can edit your message or just wait until you're "done" actually replying.

Back to wf br, you need to take their new DNS, you can find it in `Game.log`

**Comment by FRNathan13** [comment]
I don't found the dns in log
[Game.log.zip](https://github.com/Levak/warfacebot/files/1203501/Game.log.zip)

**Comment by Levak** [comment]
> I don't found the dns in log

Alright, it's not there because BR has only one realm. Next guess is in `Engine.pak/config/system.cfg` or `GameData.pak/config/system.cfg`

**Comment by FRNathan13** [comment]
I will try open CheatEngine to get new socket address like in your post "How i hacked into warface ingame protocol"

**Comment by FRNathan13** [comment]
I found "game.warface.levelupgames.com.br" in CheatEngine string table from warface.

**Comment by FRNathan13** [comment]
http://my-addr.com/check-open-ports/check-listening-of-ports/check_ports_tool.php?addr=game.warface.levelupgames.com.br&ports_selected=80&ports_list=5222%2C+5224&x=0&y=0
Timeout two possible ports.

**Comment by FRNathan13** [comment]
Without debug mode, connect fine. With debug-wb freeze at <stream:stream ... >

**Comment by FRNathan13** [comment]
```xmpp
g_language = english
game_hwid = 0
wb_accept_friend_requests = 1
wb_postpone_friend_requests = 0
wb_accept_clan_invites = 1
wb_postpone_clan_invites = 0
wb_enable_invite = 1
online_server = game.warface.levelupgames.com.br
game_version = 1.15000.789.36300
game_server_name = br

Warfacebot Copyright (C) 2015-2017 Levak Borok
This program comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions; see AGPLv3 Terms for details.


Starting TLS connection...
done.
JID: gamek01@warface/GameClient
Channel: main_pve_9
Profile ID: 3137243
Nickname: -.(MongoDB).-
Friend: [1;32m*NF_DSR*[0m
Friends: 1/50
Joined channel main_pve_9 (pve)
Experience: 6306
Rating points: 0
Money: 11445
Crowns: 0
Kredits: 0
Unlocked items: 3/115
CMD#```

**Comment by Diablo931234** [comment]
Nathan à quoi sert ce bot?

**Comment by FRNathan13** [comment]
@Diablo931234 What?

**Comment by Levak** [comment]
> Without debug mode, connect fine. With debug-wb freeze at <stream:stream ... >

What is your version of libreadline?
You seem to be using Linux emulation on Windows 10.

**Comment by FRNathan13** [comment]
Yes i'm using linux emulator. I downloaded libreadline version of travis.yml file

**Comment by Levak** [comment]
You probably want to install ubuntu packages instead of using the ones compiled for Windows.
Linux subsystem under Win 10 seem to be the wine equivalent for Linux, and under wine, you need to download Windows packages instead of the Linux host ones. So I assume Linux subsystem requires linux packages to run properly.

My guess:

```bash
$ sudo apt-get install libreadline5-dev libssl-dev libzip-dev
$ make clean debug-wb -j4
```

Make sure the packages I mention above are correct for Linux subsystem, can't get a grab on it rn.

**Comment by FRNathan13** [comment]
Same :\

Stopped at ```xml
<?xml version='1.0' ?><stream:stream to='warface' xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' xml:lang='en' version='1.0'>
```

**Comment by FRNathan13** [comment]
I Tested on debian debug-wb worked fine, without libreadline5-dev, just libreadline-gplv2-dev, but on VM is small texts xD

**Comment by FRNathan13** [comment]
I will try to compile using OSTYPE=mingw

**Comment by Levak** [comment]
Hi,

I got my hands on Linux subsystem for Windows 10 (lxwin). I can confirm the fact the bot is stuck on launch after sending the first query in debug-mode. Although, you said warfacebot was working for you in non-debug mode, and this is where I have to disagree.

To see if the problem lies in Ubuntu or in lxwin, I did the following, to at least see if it is a problem in lxwin network configuration:

```
$ echo '<?xml version='1.0' ?><stream:stream to='warface' xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' xml:lang='en' version='1.0'>' | netcat eu-comm.wf.my.com 5222
```

Debugging with `gdb` tells me that the threads seem to be working as expected, that the socket is open and that it tries to send the packet. In Wireshark, on the other hand,I see no trace of that said packet (appart from the SYN/SYN-ACK/ACK).

Narrowing it down, leads me to the flag `MSG_MORE` I give to `send(3)`. Removing the flag from `src/stream/tls.c:74`[1] (put 0 instead) solves the problem. That flag is there to force buffering inside `stream_send_msg()`, and a flush is done using `stream_flush()`. After doing little tests, it seems that on lxwin, doing `send(fd, "", 0, 0);` doesn't actually flush the stream.

After looking at Wireshark dumps without `MSG_MORE`, it seems that everything is "ok" (no extra packets, buffering is done correctly). I remember adding this flag because it was giving me troubles at first. Still unsure what to do now - probably rework that part of the code to prevent buffering. kernel-side.

[1] https://github.com/Levak/warfacebot/blob/master/src/stream/tls.c#L74
